src/lib/array.jl

```diff
@@ -570,6 +570,13 @@ end
   return (Ā,)
 end
 
+# The adjoint for exp(::AbstractArray) intercepts ChainRules' rrule for exp(::Hermitian),
+# so we call it manually. This can be removed when the generic rule for exp is moved to
+# ChainRules
+function _pullback(::AContext, ::typeof(exp), A::LinearAlgebra.RealHermSymComplexHerm)
+    return chain_rrule(exp, A)
```

Comment:
This seems to be the only special-casing in this file, so if anyone knows a better way to work around the generality of the current `exp` adjoint, let me know.

Reply:
Do we need to do it via `_pullback`? Seems like a regular `@adjoint` should do the trick too

Reply:
We don't need to do it via `_pullback`, it just let me use `chain_rrule` directly. The alternative is this:

```suggestion
@adoint exp(A::LinearAlgebra.RealHermSymComplexHerm)
    Y, back = chain_rrule(exp, A)
    return Y, Δ -> (back(Δ)[2],)
```

Reply:
Yeah that's what I was thinking, thanks